docs/nebula-design-concepts.md

```diff
@@ -0,0 +1,213 @@
+the same edge type.
+These edges will be differentiated by their rankings. So, any
+edge is uniquely identified by the tuple [src_vertex, dst_vertex, edge_type, ranking]. In
+the preceding example, edges are **serve**
+and **likeness**.
+
+### Tags
+
+In **Nebula Graph**, vertex properties are clustered by **tags**. A tag is like an entity type, which contains multiple properties. Each vertex can have multiple tags associated. There could be dependency between tags, too.
+
+For instance, we could have a tag **Person**, which contains first_name, last_name, date_of_birth, etc. We could have another tag **Developer**, which has properties favorite_lang, github_account, etc. Tag **Developer** would depend on tag **Person**, which means if a vertex has tag **Developer** associated, it must have tag **Person** associated as well.
+
+
+### Properties
+
+Both tags and edges can have properties, which are key/value pairs, and
+the definition of these properties (name, type, etc.) are called **Schema** in the **Nebula Graph**. Like Mysql, **Nebula Graph** is a strong typed database. The name and data type of the properties are determined before the data is written. Vertices can have properties indirectly via associated tags, and edges can have properties too.
+
+In the preceding example, the schema of the tag **player** has three kind of properties:
+
+- id (vid)
+- Name (string)
+- Age (int)
+
+ the schema of the edge **serve** has two kinds of properties:
+
+ - start_year (int)
+ - end_year (int)
+
+
+### Edge type
+
+Each edge is an instance of an edge type. Each edge type contains a collection of properties (name, type, etc.).
+
+This part is written based on my own understand.
+-->
+
+
+### Vertex, vertex tag and tag properties
+
+In **Nebula Graph**, the type of vertex is called **tag**. **A vertex must be labeled with at least one tag. Each tag may have a set of properties called schema.**
+
+Consider the above picture as example, there are two tags: **player** and **team**. Player has three properties: id (vid), Name(string), and Age(int); team has two properties: id (vid) and Name(string).
+
+Like Mysql, **Nebula Graph** is a strong typed database. The property name and data type are determined before the data is written.
+
+### Edge, edge type and the corresponding edge properties
+
+**Edges in Nebula Graph always have a type, a start (src)and an end vertex (dst), and a direction.** The direction is implicitly defined by the source and destination vertices. Edges can be self-referencing/looping but can never be dangling (missing a start or end vertex). Each edge in **Nebula Graph** is always of an edge type, has a source and a destination vertices, and carries a ranking number (optional).
```

Comment:
Edges can be self-referencing/looping /* but can never be dangling (missing a start or end vertex) */

comment out